composer/trainer/trainer.py

```diff
@@ -1020,9 +1099,248 @@ def saved_checkpoints(self) -> List[Tuple[Timestamp, List[pathlib.Path]]]:
+            self.state.max_duration = _scale_max_duration_by_ssr(scale_schedule_ratio, self.state.max_duration)
+        if schedulers is not None:
+            self.state.schedulers = _compile_schedulers(schedulers, self.state, scale_schedule_ratio)
+
+            if step_schedulers_every_batch is None:
+                self._scheduler_step_frequency = _get_default_scheduler_frequency(schedulers)
+            else:
+                self._scheduler_step_frequency = TimeUnit.BATCH if step_schedulers_every_batch else TimeUnit.EPOCH
+        else:
+            if scale_schedule_ratio != 1.0:
+                raise ValueError("Specifying `scale_schedule_ratio` without `schedulers` has no effect.")
+
+            if step_schedulers_every_batch is not None:
+                raise ValueError("Specifying `step_schedulers_every_batch` without `schedulers` has no effect.")
+
+            if step_schedulers_every_batch is not None:
+                raise ValueError("Specifying `step_schedulers_every_batch` without `schedulers` has no effect.")
+
+        if len(ensure_tuple(schedulers)) == 0:
+            warnings.warn(f"NoSchedulerWarning: No schedulers were specified. The learning rate will be constant.")
+
+        # Evaluators
+        if eval_dataloader is not None:
+            evaluators = [
+                # Need to use the `original_model` rather than `state.model`, as `state.model`
+                # could be DDP / DeepSpeed wrapped.
+                ensure_evaluator(evaluator, default_metrics=self._original_model.metrics(train=False))
+                for evaluator in ensure_tuple(eval_dataloader)
+            ]
+            _set_evaluator_interval_and_subset_num_batches(
+                evaluators=evaluators,
+                eval_interval=eval_interval,
+                subset_num_batches=eval_subset_num_batches,
+            )
+            if len(evaluators) == 0:
+                if eval_subset_num_batches != -1:
+                    raise ValueError("Specifying `eval_subset_num_batches` without an `eval_dataloader` has no effect.")
+                if eval_interval != 1:
+                    raise ValueError("Specifying `eval_interval` without an `eval_dataloader` has no effect.")
+
+            self.state.evaluators = evaluators
+
+        if len(self.state.evaluators) == 0:
+            warnings.warn(("No `eval_dataloader` was specified. Please specify `eval_dataloader` to periodically "
+                           "evaluate your model while training."))
+
+        # Grad Accum
+        if grad_accum is not None:
+            self.adaptive_gradient_accumulation = _is_adaptive_grad_accum(grad_accum, device=self._device)
+            self.state.grad_accum = _get_initial_grad_accum(grad_accum)
+
+        # Grad Clip Norm
+        if grad_clip_norm is not None:
+            if self.deepspeed_enabled:
+                raise ValueError("Changing the grad_clip_norm when using DeepSpeed is not supported.")
```

Comment:
Maybe this should just be a warning?

Reply:
If the user specifies this value (i.e. not None), and we aren't able to apply it, then I would prefer to keep this an error, since it is easy to miss over a warning.